README.tmpl.md

```diff
@@ -0,0 +1,206 @@
+information to reduce the rate of false positive spelling errors where
+words refer to labels within the source code.
+
+
+## Installation
+
+Beyond the standard Go installation process, you must also have libhunspell
+and its header files on your system. For a debian-based system this is done
+with `sudo apt install libhunspell-dev`.
+
+
+## Work Flow
+
+`gospel` makes use of module information, and so must be run within a module
+and will only consider source files that are referred to by the module.
+
+`gospel` can be used without configuration in many cases, particularly for
+small source trees where the number of potential errors is likely to be
+small. However, its behaviour can be tuned with [configuration files](#configuration-files).
+The initial setup workflow makes use of the [`.words`](#.words) file to
+build a dictionary of candidate words to ignore.
+
+```
+$ gospel -misspellings=.words ./...
+```
+
+This will collate all the found misspellings in the source tree and write
+them to `.words`. If `gospel` is run again, it will accept the all words in
+the source tree as correctly spelled as they now appear in the projects
+dictionary. Since hunspell dictionaries are just text files, you can go
+through the list of words to remove properly identified misspellings and
+leave in words that are domain-specific and incorrectly marked.
+
+For example the `.words` file at the root of the `gospel` repo includes
+words that would otherwise be flagged with the default "en_US" spelling
+dictionary and was autogenerated using the command above.
+
+```
+{{.words -}}
+```
+
+This file can also be edited to make use of more advanced hunspell
+features. See [Hunspell Dictionaries](#hunspell-dictionaries) below.
+
+
+## Command Line Options
+
+`gospel` provides a number of command line options. The full list can
+be obtained by running `gospel -h`, but the majority correspond directly
+with `.gospel.conf` options and are described [there](.gospel.conf).
+
+The remaining options are not intended to be persistently stored:
+
+- `-config` — whether to use config file (default true, intended for debugging use).
+- `-dict-paths` — a colon-separated directory list containing hunspell dictionaries (defaults to a system-specific value).
+- `-misspellings` — a file path to write a dictionary of misspellings to (see [Work Flow](#work-flow) above).
+- `-since` — a git ref specifying that only changes since then should be considered for misspelling (requires git).
+- `-update-dict` — whether the `-misspellings` flag is being used to update a dictionary that already exists.
+- `-write-config` — emit a config file based on flags and existing config to stdout and exit.
+
+
+## Configuration Files
+
+`gospel` uses two configuration file types, `.words` files at the module
+roots of packages that are being checked, and a `.gospel.conf` file at the
+module root of the module in which `gospel` was invoked.
```

Comment:
thinking aloud for a second: perhaps worth thinking about the workspace root (the new Go 1.18 feature) as a fallback for finding `.gospel.conf`, or even `.words`.

Reply:
Yes, it would be nice to handle some of this more gracefully. Can you point to docs?

Reply:
https://go.dev/ref/mod#workspaces